facebook/OpenBIC
# yv4: sd: revise the register for checking BMC interface

# Description:
Revise the register of SD CPLD for checking the interface to communicate with BMC.

# Motivation:
SD CPLD team help to revise the register so that the value of the bit won't be wrong if using the SD CPLD firmware that doesn't support the bit for selecting BMC interface.

# Test plan:
1. Check that BIC will use I2C interface to communicate with BMC if the SD CPLD firmware doesn't support the bit for selecting BMC interface.

2. Change to use I3C interface to communicate with BMC successfully.

# Test logs:
1. Tested with the SD CPLD firmware doesn't support the bit for selecting BMC interface and check that BIC will use I2C interface to communicate with BMC.

- BIC console uart:~$ [00:00:05.125,000] <inf> util_sys: BIC Ready [00:00:05.125,000] <inf> plat_gpio: Reset USB Hub
[00:00:05.125,000] <inf> plat_mctp: Using I2C interface to communicate with BMC

2. Change to use I3C interface to communicate with BMC successfully. 1. Use the following steps to change the interface to I3C (test on slot5). 1. Power off host. 2. Stop pldmd and mctpd services. 3. Set SD CPLD IOE root@bmc:~# gpioset gpiochip5 24=1 4. Reset BIC - Check that BIC is using I2C interface after SD BIC reset uart:~$ [00:00:05.125,000] <inf> util_sys: BIC Ready [00:00:05.125,000] <inf> plat_gpio: Reset USB Hub [00:00:05.126,000] <inf> plat_mctp: Using I3C interface to communicate with BMC 5. BMC enable hot join to detect SD BIC I3C device. 6. Start mctpd and pldmd services. 7. BMC manual set up the mctp-i3c link. root@bmc:~# mctp link set mctpi3c1 up root@bmc:~# mctp addr add 8 dev mctpi3c1 root@bmc:~# busctl call xyz.openbmc_project.MCTP /xyz/openbmc_project/mctp au.com.CodeConstruct.MCTP LearnEndpoint say "mctpi3c1" 6 0x07 0xec 0x80 0x01 0x00 0x14 8. Power on host.

    2. Check BMC could communicate with BIC through PLDM over MCTP over I2C
            root@bmc:~# pldmtool base GetTID -m 50
            {
                    "Response": 50
            }
